Tests/scripts/collect_tests_and_content_packs.py

```diff
@@ -1241,31 +1285,71 @@ def get_test_list_and_content_packs_to_install(files_string,
     packs_to_install = packs_to_install.union(packs_of_collected_tests)
 
     # All filtering out of packs should be done here
-    packs_to_install = filter_installed_packs(packs_to_install)
+    packs_to_install = filter_installed_packs(packs_to_install, marketplace_version, id_set)
 
     # All filtering out of tests should be done here
-    tests = filter_tests(tests, id_set, modified_packs)
+    tests = filter_tests(tests, id_set, modified_packs, marketplace_version, False)
 
-    if not tests or changed_common:
-        if not tests:
-            logging.info("No tests found running sanity check only.")
-        else:
-            logging.info("Changed one of the Common Server files, running sanity check too.")
+    if marketplace_version != 'marketplacev2':
+        if not tests or changed_common:
+            if not tests:
+                logging.info("No tests found running sanity check only.")
+            else:
+                logging.info("Changed one of the Common Server files, running sanity check too.")
 
-        logging.debug(f"Adding sanity tests: {SANITY_TESTS}")
-        tests.update(SANITY_TESTS)
-        logging.debug("Adding HelloWorld to tests as most of the sanity tests requires it.")
-        logging.debug(
-            "Adding Gmail to packs to install as 'Sanity Test - Playbook with Unmockable Integration' uses it"
-        )
-        packs_to_install.update(["HelloWorld", "Gmail"])
+            logging.debug(f"Adding sanity tests: {SANITY_TESTS}")
+            tests.update(SANITY_TESTS)
+            logging.debug("Adding HelloWorld to tests as most of the sanity tests requires it.")
+            logging.debug(
+                "Adding Gmail to packs to install as 'Sanity Test - Playbook with Unmockable Integration' uses it"
+            )
+            packs_to_install.update(["HelloWorld", "Gmail"])
 
     # We add Base and DeveloperTools packs for every build
-    packs_to_install.update(["DeveloperTools", "Base"])
+    if tests or packs_to_install:
+        packs_to_install.update(["DeveloperTools", "Base"])
 
     return tests, packs_to_install
 
 
+def get_all_packs_with_artifacts_for_specific_marketplace(id_set: dict, marketplace_version: str):
+    """Get all packs that have content artifacts in specific marketplace version:
+        Args:
+            id_set (dict): The id set object
+            marketplace_version (str): The marketplace version were the tests will run
+
+        Return:
+             set: set of packs
+        """
+    packs_to_install = set()
+    for artifacts in id_set.values():
+        # Ignore the Packs list in the ID set
+        if isinstance(artifacts, dict):
+            break
```

Comment:
I think this should be "continue" no? we don't want to exit the loop if we get to the packs entry but to skip it.
```suggestion
            continue
```